psh/pshapp/pshapp.c

```diff
@@ -1136,9 +1139,38 @@ static int psh_runscript(char *path)
 			free(argv);
 			argv = NULL;
 		}
+		else {
+			for (size_t j = 0; j < sizeof(scriptCmds) / sizeof(scriptCmds[0]); j++) {
+				size_t len = strlen(scriptCmds[j]);
+
+				if ((line[len] != '\0') && (isspace(line[len]) == 0) && (strncmp(line, scriptCmds[j], len) != 0)) {
+					continue;
+				}
```

Comment:
You merged this PR, but I have small comment.
If the length of the variable `line` is `< len`, values outside the range of the read `line` are checked.
Solution: move `strncmp()` to the begining of the `if` statement, ie:
```c
				if ((strncmp(line, scriptCmds[j], len) != 0)) && (line[len] != '\0') && (isspace(line[len]) == 0))  {
					continue;
				}
```
or use `ret` value ( which is length of the `line` ).
Btw.
allocating and releasing memory for each line loaded is not inefficient.

Reply:
Thanks, fix in [PR](https://github.com/phoenix-rtos/phoenix-rtos-utils/pull/206). I decided to use `ret` value (in fixing PR `ret` becomes `lineLen`), also removed inefficient memory releasing as suggested because `getline()` supports memory reallocs.